p2p/pex/pex_reactor_test.go

```diff
@@ -35,6 +37,77 @@ func TestPEXReactorBasic(t *testing.T) {
 	assert.NotEmpty(t, r.GetChannels())
 }
 
+func TestPEXReactorStopPeerWithOutInitPeer(t *testing.T) {
```

Comment:
This test doesn't always fail when `InitiPeer` is removed.

Reply:
It sometimes failed with `InitPeer`.

Reply:
The issue I fixed is also not always happen. It depends on the schedule of goroutine,  as I describe:
![image](https://user-images.githubusercontent.com/7310198/56455537-f696da80-6391-11e9-8744-6532e0b9ebe3.png) 
this is happen sometimes.  That is why the case run 10000 times to make it happen.

Reply:
If the success of the fix is sensistive to the inner workings of the runtme we need to find a way to never fail, no matter how the routines are scheduled.